qcodes_contrib_drivers/drivers/SignalCore/SignalCore.py

```diff
@@ -0,0 +1,311 @@
+                ('ext_ref_detect', ctypes.c_uint8),
+                ('ref_out_select', ctypes.c_uint8),
+                ('list_mode_running', ctypes.c_uint8),
+                ('rf_mode', ctypes.c_uint8),
+                ('over_temp', ctypes.c_uint8),
+                ('harmonic_ss', ctypes.c_uint8),
+                ('pci_clk_enable', ctypes.c_uint8)]
+
+
+class DeviceStatusT(ctypes.Structure):
+    _fields_ = [('list_mode_t', ListModeT),
+                ('operate_status_t', OperateStatusT),
+                ('pll_status_t', PLLStatusT)]
+device_status_t = DeviceStatusT()
+
+
+class HWTriggerT(ctypes.Structure):
+    _fields_ = [('edge', ctypes.c_uint8),
+                ('pxi_enable', ctypes.c_uint8),
+                ('pxi_line', ctypes.c_uint8)]
+hw_trigger_t = HWTriggerT()
+
+
+class DeviceRFParamsT(ctypes.Structure):
+    _fields_ = [('frequency', ctypes.c_double),
+                ('sweep_start_freq', ctypes.c_double),
+                ('sweep_stop_freq', ctypes.c_double),
+                ('sweep_step_freq', ctypes.c_double),
+                ('sweep_dwell_time', ctypes.c_uint32),
+                ('sweep_cycles', ctypes.c_uint32),
+                ('buffer_points', ctypes.c_uint32),
+                ('rf_phase_offset', ctypes.c_float),
+                ('power_level', ctypes.c_float),
+                ('atten_value', ctypes.c_float),
+                ('level_dac_value', ctypes.c_uint16)]
+device_rf_params_t = DeviceRFParamsT()
+
+
+error_dict = {'0':'SCI_SUCCESS',
+              '0':'SCI_ERROR_NONE',
+              '-1':'SCI_ERROR_INVALID_DEVICE_HANDLE',
+              '-2':'SCI_ERROR_NO_DEVICE',
+              '-3':'SCI_ERROR_INVALID_DEVICE',
+              '-4':'SCI_ERROR_MEM_UNALLOCATE',
+              '-5':'SCI_ERROR_MEM_EXCEEDED',
+              '-6':'SCI_ERROR_INVALID_REG',
+              '-7':'SCI_ERROR_INVALID_ARGUMENT',
+              '-8':'SCI_ERROR_COMM_FAIL',
+              '-9':'SCI_ERROR_OUT_OF_RANGE',
+              '-10':'SCI_ERROR_PLL_LOCK',
+              '-11':'SCI_ERROR_TIMED_OUT',
+              '-12':'SCI_ERROR_COMM_INIT',
+              '-13':'SCI_ERROR_TIMED_OUT_READ',
+              '-14':'SCI_ERROR_INVALID_INTERFACE'}
+
+
+
+class SC5521A(Instrument):
+    __doc__ = 'QCoDeS python driver for the Signal Core SC5521A.'
+
+    def __init__(self, name: str,
+                       dll_path: str='SignalCore\\SC5520A\\api\\c\\scipci\\x64\\sc5520a_uhfs.dll',
+                       **kwargs):
+        """
+        QCoDeS driver for the Signal Core SC5521A.
+        This driver has been tested when only one SignalCore is connected to the
+        computer.
+
+        Args:
+        name (str): Name of the instrument.
+        dll_path (str): Path towards the instrument DLL.
+        """
+        
+        (super().__init__)(name, **kwargs)
+        
+        # Adapt the path to the computer language
+        dll_path = os.path.join(os.environ['PROGRAMFILES'], dll_path)
+
+        self._dll = ctypes.WinDLL(dll_path)
```

Comment:
so let's use this https://mypy.readthedocs.io/en/stable/common_issues.html#python-version-and-system-platform-checks here - and if the platform is not windows then raise a nice error to the user (mypy will also like it and understand it)

Reply:
Nice, I am implementing it, thanks!

Reply:
should be done now